common/remote/sockfile.cpp

```diff
@@ -4003,6 +4011,10 @@ class CRemoteFileServer : public CInterface, implements IRemoteFileServer
         // can revert to previous behavior with conf file setting "allow_pgcache_flush=false"
         if (extraFlags == IFEnone)
             extraFlags = IFEnocache;
+        // also try to recv extra flags
+        unsigned eflags = IFUnone;
+        if (msg.remaining() >= sizeof(unsigned))
```

Comment:
Would also be good if serialized if shifted and serialized as unsigned short here.